### PR TITLE
chore: Upgrades `privatelink_endpoint_service_serverless` resource to auto-generated SDK

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Upgrades `privatelink_endpoint_service_serverless` resource to auto-generated SDK
Add migration test

Link to any related issue(s): CLOUDP-226089

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go [hunk 1]
```diff
@@ -76,18 +76,12 @@ func PluralDataSource() *schema.Resource {
...
 
-func dataSourceMongoDBAtlasPrivateLinkEndpointsServiceServerlessRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	// Get client connection.
-	conn := meta.(*config.MongoDBClient).Atlas
+func dataSourcePluralRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
 	projectID := d.Get("project_id").(string)
 	instanceName := d.Get("instance_name").(string)
 
-	options := &matlas.ListOptions{
```
Comment (oarbusi) [inline internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go:85]: no more need of pagination options, endpoint called by new SDK returns all serverless private endpoints

File: internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoint_service_serverless.go [hunk 2]
```diff
@@ -62,48 +62,43 @@ func DataSource() *schema.Resource {
...
 
-	if err := d.Set("status", serviceEndpoint.Status); err != nil {
+	if err := d.Set("status", serviceEndpoint.GetStatus()); err != nil {
 		return diag.FromErr(fmt.Errorf(privatelinkendpointservice.ErrorEndpointSetting, "status", endpointID, err))
 	}
 
-	if err := d.Set("comment", serviceEndpoint.Comment); err != nil {
+	if err := d.Set("comment", serviceEndpoint.GetComment()); err != nil {
 		return diag.FromErr(fmt.Errorf(privatelinkendpointservice.ErrorEndpointSetting, "comment", endpointID, err))
 	}
```
Comment (lantoli) [inline internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoint_service_serverless.go:88]: no ErrorMessage?

File: internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go [hunk 3]
```diff
@@ -76,18 +76,12 @@ func PluralDataSource() *schema.Resource {
...
-		PageNum:      d.Get("page_num").(int),
```
Comment (lantoli) [inline internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go:86]: should we deprecate page_num and iterms_per_page ?

File: internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go [hunk 4]
```diff
@@ -101,7 +95,7 @@ func dataSourceMongoDBAtlasPrivateLinkEndpointsServiceServerlessRead(ctx context
 	return nil
 }
 
-func flattenServerlessPrivateLinkEndpoints(privateLinks []matlas.ServerlessPrivateEndpointConnection) []map[string]any {
+func flattenServerlessPrivateLinkEndpoints(privateLinks []admin.ServerlessTenantEndpoint) []map[string]any {
 	var results []map[string]any
```
Comment (lantoli) [inline internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go]: I think it's more clear constructions like:

var results []map[string]any

if len(privateLinks) == 0 {
   return results
}
results = make([]map[string]any, len(privateLinks))


to be like this:

if len(privateLinks) == 0 {
  return nil
}

results := make([]map[string]any, len(privateLinks))

File: internal/service/privatelinkendpointserviceserverless/resource_privatelink_endpoint_service_serverless.go [hunk 5]
```diff
@@ -279,34 +270,35 @@ func resourceMongoDBAtlasPrivateLinkEndpointServiceServerlessImportState(ctx con
 	return []*schema.ResourceData{d}, nil
 }
 
-func resourceServiceEndpointServerlessRefreshFunc(ctx context.Context, client *matlas.Client, projectID, instanceName, endpointServiceID string) retry.StateRefreshFunc {
+func resourceServiceEndpointServerlessRefreshFunc(ctx context.Context, client *admin.APIClient, projectID, instanceName, endpointServiceID string) retry.StateRefreshFunc {
```
Comment (lantoli) [inline internal/service/privatelinkendpointserviceserverless/resource_privatelink_endpoint_service_serverless.go]: rename to something like resourceRefreshFunc

File: internal/service/privatelinkendpointserviceserverless/resource_privatelink_endpoint_service_serverless.go [hunk 6]
```diff
@@ -279,34 +270,35 @@ func resourceMongoDBAtlasPrivateLinkEndpointServiceServerlessImportState(ctx con
...
 		}
+		resultStatus := serverlessTenantEndpoint.GetStatus()
 
-		return i, i.Status, nil
+		return serverlessTenantEndpoint, resultStatus, nil
 	}
 }
 
-func resourceServerlessInstanceListRefreshFunc(ctx context.Context, projectID string, client *matlas.Client) retry.StateRefreshFunc {
+func resourceServerlessInstanceListRefreshFunc(ctx context.Context, projectID string, client *admin.APIClient) retry.StateRefreshFunc {
```
Comment (lantoli) [inline internal/service/privatelinkendpointserviceserverless/resource_privatelink_endpoint_service_serverless.go]: rename to something like resourceListRefreshFunc

File: internal/service/privatelinkendpointserviceserverless/resource_privatelink_endpoint_service_serverless_migration_test.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+		datasourceEndpointsName = "data.mongodbatlas_privatelink_endpoints_service_serverless.test"
+		orgID                   = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName             = acctest.RandomWithPrefix("test-acc-serverless")
+		instanceName            = acctest.RandomWithPrefix("test-acc-serverless")
+		commentOrigin           = "this is a comment for serverless private link endpoint"
+		config                  = configBasic(orgID, projectName, instanceName, commentOrigin)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acc.PreCheckBasic(t) },
```
Comment (lantoli) [inline internal/service/privatelinkendpointserviceserverless/resource_privatelink_endpoint_service_serverless_migration_test.go]:
```suggestion
		PreCheck:     func() { mig.PreCheckBasic(t) },
```

File: internal/service/privatelinkendpointserviceserverless/resource_privatelink_endpoint_service_serverless_migration_test.go [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+		CheckDestroy: checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: mig.ExternalProviders(),
+				Config:            config,
+				Check: resource.ComposeTestCheckFunc(
+					checkExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "provider_name", "AWS"),
+					resource.TestCheckResourceAttr(resourceName, "comment", commentOrigin),
+					resource.TestCheckResourceAttr(datasourceName, "comment", commentOrigin),
```
Comment (lantoli) [inline internal/service/privatelinkendpointserviceserverless/resource_privatelink_endpoint_service_serverless_migration_test.go:34]: i would only check resourceName in mig test

**Review by lantoli — APPROVED** [review]

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoint_service_serverless.go (hunk 2 above)
Comment (oarbusi) [reply internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoint_service_serverless.go:88]: Not sure I understand what you mean, can you explain it?

File: internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go (hunk 3 above)
Comment (oarbusi) [reply internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go:86]: I would leave this for future project to remove pagination in different resources and avoid breaking changes for now.

File: internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoint_service_serverless.go (hunk 2 above)
Comment (lantoli) [reply internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoint_service_serverless.go:88]: if err := d.Set("error_message", serviceEndpoint.ErrorMessage); err != nil { ...

was deleted but not replaced

File: internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go (hunk 3 above)
Comment (lantoli) [reply internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go:86]: i mean to flag the properties as deprecated, I think we should do it here as we're doing with other data sources.
Not saying to sunset/delete them here, that will be later and the real breaking change.

File: internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoint_service_serverless.go (hunk 2 above)
Comment (oarbusi) [reply internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoint_service_serverless.go:88]: Now I understand what you mean. Error message was duplicated here so I deleted one of them

File: internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go (hunk 3 above)
Comment (oarbusi) [reply internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go:86]: Understood! I set them as Deprecated in the latest commit

Comment (lantoli) [reply internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go:86]: great.  Can you better use or create a new constant in deprecation.go file in constant package, so it's easier to search for deprecated params, like constant.DeprecationParamByDateWithReplacement ...  , etc. ?

Comment (oarbusi) [reply internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go:86]: Done!

**Pull request merged**